docs/best-practice-consensus-pipeline.md

```diff
@@ -0,0 +1,217 @@
+<img src="https://github.com/fulcrumgenomics/fgbio/blob/main/.github/logos/fulcrumgenomics.svg" alt="Fulcrum Genomics" height="35"/>
+
+# fgbio Best Practise FASTQ->Consensus Pipeline
+
+This document describes the best practise(s) for building pipelines to go from FASTQ files that contain a mixture of template (genomic) bases and UMI bases, through to aligned and filtered consensus reads.  
+
+The pipeline is broken into two phases:
+
+1. Getting to aligned and grouped raw reads
+2. Going from raw reads to aligned and filtered consensus reads
+
+We provide two alternatives for the second phase.  One intended for R&D environment where it is desirable to be able to branch off from the pipeline and test e.g. multiple consensus calling or filtering parameters.  The other is intended for high throughput production environments where performance and throughput take precedence over flexibility.
+
+## Tools used
+
+The steps described below use a mixture of:
+
+- `fgbio` (version 2.0 or higher)
+- `samtools` (version 
+- `bwa mem`
+
```

Comment:
commenting here, but I think we should do two things:

1. use `--compression 1` when writing to disk, and `--compression 0` when piping
2. highlight somewhere `--async-io=true`, since we omit that in a few places